python.d/haproxy.chart.py

```diff
@@ -0,0 +1,167 @@
+# -*- coding: utf-8 -*-
+# Description: haproxy netdata python.d module
+# Author: Pawel Krupa (paulfantom)
+
+from base import SocketService, UrlService
+import csv
+
+# default module values (can be overridden per job in `config`)
+# update_every = 2
+priority = 60000
+retries = 60
+
+# charts order (can be overridden if you want less charts, or different order)
+ORDER = ['qcur', 'scur', 'bin', 'bout']
+POSITION = [2, 4, 8, 9]
+
+CHARTS = {
+    'qcur': {
+        'options': ["", "Current queue", 'per sec', '', '', 'line'],
+        'lines': [
+            ['name', None, 'incremental']
+        ]},
+    'scur': {
+        'options': ["", "Current session rate", 'per sec', '', '', 'line'],
+        'lines': [
+            ['name', None, 'incremental']
+        ]},
+    'bin': {
+        'options': ["", "Bytes in", 'kilobytes/s', '', '', 'line'],
+        'lines': [
+            ['name', None, 'incremental', 1, 1024]
+        ]},
+    'bout': {
+        'options': ["", "Bytes out", 'kilobytes/s', '', '', 'line'],
+        'lines': [
+            ['name', None, 'incremental', 1, 1024]
+        ]},
+}
+
+
+class Service(SocketService, UrlService):
+    def __init__(self, configuration=None, name=None):
+        self.use_socket = None
+        if 'unix_socket' in configuration:
+            SocketService.__init__(self, configuration=configuration, name=name)
+            self.request = "show stat\r\n"
+            self.use_socket = True
+        else:
+            UrlService.__init__(self, configuration=configuration, name=name)
+            if not self.url.endswith(";csv;norefresh"):
+                self.url += ";csv;norefresh"
+            self.use_socket = False
+
+        # self.order and self.definitions are created with _create_definitions method
+        # self.order = ORDER
+        # self.definitions = CHARTS
+        self.order = []
+        self.definitions = {}
+
+    def _get_parsed_data(self):
+        """
+        Retrieve and parse raw data
+        :return: dict
+        """
+        try:
+            if self.use_socket:
+                raw = SocketService._get_raw_data(self)
+            else:
+                raw = UrlService._get_raw_data(self)
+        except (ValueError, AttributeError):
+            return None
+
+        try:
+            return [row for row in csv.reader(raw.splitlines(), delimiter=',')]
+        except Exception as e:
+            self.debug(str(e))
+            return None
+
+    def _get_data(self):
+        """
+        Format data
+        :return: dict
+        """
+        parsed = self._get_parsed_data()
+        if parsed is None or len(parsed) == 0:
```

Comment:
if `parsed` is valid, it's a non-empty list.  Otherwise it's either None or an empty list, both of which are false-y.  So just do:

    if not parsed: